dartagnan/src/main/java/com/dat3m/dartagnan/program/event/lang/pthread/Lock.java

```diff
@@ -31,21 +31,17 @@ public String defaultString() {
         return "pthread_mutex_lock(&" + name + ")";
     }
 
+    //TODO: Why does this have a ResultRegister without inheriting from RegWriter?
+    // Why is this even needed?
```

Comment:
It is used in the compilation to create the corresponding Load. We should probably make it inherit from `RegWriter`

Reply:
But the register a pure dummy register, no? The compilation can simply create a fresh one instead of using some pre-defined one.

Reply:
You are right. We can fully remove the register from lock/unlock events.